.github/workflows/balena.yml

```diff
@@ -122,3 +127,32 @@ jobs:
               ((retry_count++))
             fi
           done
+          echo "::set-output name=sut-container-result::${result}"
+
+      - name: Stream logs from SUT container
+        id: sut-container-logs
+        if: ${{ github.event_name == 'pull_request' && github.event.action != 'closed'}}
+        run: |
+          set -uae
+
+          [[ '${{ env.VERBOSE }}' =~ on|On|Yes|yes|true|True ]] && set -x
+          balena login --token '${{ secrets.BALENACLOUD_API_KEY }}'
+          balena whoami
+
+          balena logs ${{ steps.find-test-device.outputs.balena_dut_uuid }} --service sut
+
+      - name: Mark release result
+        id: result-check
+        if: ${{ github.event_name == 'pull_request' && github.event.action != 'closed'}}
+        run: |
+          set -uae
+
+          if [ "${{env.sut-container-result}}" = "0" ] ; then
+            exit 0
+          fi
+
+          echo "Release failed"
+          exit 1 
```

Comment:
What do you need this step for?

Reply:
We need to pass or fail the GitHub action to indicate explicitly our test result. So if the test result failed then the GitHub action should fail.

Reply:
Oh I see you have added exit 0 to line 124 already. Cool. Will remove this.